test/validate_no_rule_description_references.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/rules.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('check for rule message descriptions in tests:', () {
+    registerLintRules();
+    for (var rule in Analyzer.facade.registeredRules) {
+      test(rule.name, () async {
+        var result = Process.runSync('grep', ['-R', rule.description, 'test']);
```

Comment:
I think this won't work for the lints that have messages encoded in custom instances of `LintCode`, nor will it work for messages that have placeholders (like `{0}`) in them.

Reply:
Good point!  This caught a bunch of bad integration tests so I'm inclined to think the coverage is useful (if incomplete).  I'll do a bit of looking at the reflective tests and see if there's sanity checking we can do there...  Thanks!

Reply:
(FWIW I see one maybe problematic case in a reflective test but not sure it's worth removing.)

Reply:
Another argument in favor of landing this is that it would have flagged all the cases in #3536.

Reply:
> I'll do a bit of looking at the reflective tests ...

My comment wasn't related to reflective tests, so I'm confused.

I just meant that (a) instead of using `rule.description` we probably need to iterate over `rule.lintCodes` and grab the `problemMessage` from each code, and (b) replace placeholders with a regexp like `.*`.

> Another argument in favor of landing this ...

Landing it without fixing these issues is fine, but we should add a comment so that our future selves can remember what's going on here. :-)